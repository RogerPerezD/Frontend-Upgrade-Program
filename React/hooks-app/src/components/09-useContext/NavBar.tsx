import React from 'react';
import { Link } from "react-router-dom";

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className='navbar-brand' to="/">useContext</Link>

                <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <Link className="nav-link active" to="/">Home</Link>
                    <Link className="nav-link" to="about">About</Link>
                    <Link className="nav-link" to="login">Login</Link>
                </ul>
                </div>
            </div>
        </nav>
  );
};
