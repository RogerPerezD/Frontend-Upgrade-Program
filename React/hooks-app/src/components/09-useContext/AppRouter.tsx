import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';
import { NotFoundScreen } from './NotFoundScreen';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route path="/"  element={ <HomeScreen />}/>
            <Route path="/about"  element={ <AboutScreen />}/>
            <Route path="/login"  element={ <LoginScreen />}/>
            <Route path="*"  element={ <NotFoundScreen />}/>
          </Routes>
        </div>
      </div>
    </Router>
  );
};
