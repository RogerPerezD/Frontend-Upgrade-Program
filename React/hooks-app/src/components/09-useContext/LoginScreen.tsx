import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

    const user= {
    id: 123,
    name: 'Roger',
    email: 'roger@mail.com'
  }

  return (
    <div>
      <h1>Login Screen</h1>
      <button 
      className='btn btn-primary'
      onClick={ ()=> { setUser( user )}}
      >
        Login
      </button>
      <hr />
    </div>
  );
};
