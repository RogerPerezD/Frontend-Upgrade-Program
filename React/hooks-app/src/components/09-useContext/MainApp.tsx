import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export interface User{
  id: number;
  name: string;
  email: string;
}

export const MainApp = () => {

  const [user, setUser] = useState<User>({} as User);


  return (
      <UserContext.Provider value={ {
        user,
        setUser
      } }>
        <AppRouter/>
      </UserContext.Provider>
      
  );
};
