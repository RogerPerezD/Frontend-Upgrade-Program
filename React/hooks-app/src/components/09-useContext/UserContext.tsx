import { createContext } from "react";
import { User } from "./MainApp";

type UserContexType = {
    user: User,
    setUser: React.Dispatch<React.SetStateAction<User>>
}


export const UserContext = createContext<UserContexType>({} as UserContexType);