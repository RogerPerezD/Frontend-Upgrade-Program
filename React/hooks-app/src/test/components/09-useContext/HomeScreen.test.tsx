import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Test in <HomeScreen/>', () => {
  
    
    test('should show the component successfully', () => {
        const wrapper = shallow( <HomeScreen/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the user in the component', () => {
        const user = {
            id: 1,
            name: 'Roger',
            email: 'roger@mail.com',
        }
        const setUser = jest.fn();
        const wrapper = mount( 
        <UserContext.Provider value={ { user, setUser} }>  
            <HomeScreen/>
        </UserContext.Provider> );

        const pre = wrapper.find('HomeScreen').find('pre').text();
        expect( wrapper ).toMatchSnapshot();
        expect( pre ).toEqual( JSON.stringify(user,null,2) );
    });
    
    
});
