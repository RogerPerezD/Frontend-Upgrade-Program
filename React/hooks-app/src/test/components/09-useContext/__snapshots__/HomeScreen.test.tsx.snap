// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HomeScreen/> should show the component successfully 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre />
</div>
`;

exports[`Test in <HomeScreen/> should show the user in the component 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
  "id": 1,
  "name": "Roger",
  "email": "roger@mail.com"
}
  </pre>
</div>
`;
